File: Java/214_ShortestPalindrome.java
class Solution {

    boolean isPali(String s, int l, int r) {
            while (l <= r) {
                if (s.charAt(l) != s.charAt(r)) {
                    return false;
                }
                l++;
                r--;
            }
            return true;
        }

    public String shortestPalindrome(String s) {
                 
        // Bucle que recorre la cadena desde el final
        for (int r = s.length() - 1; r >= 0; r--) {
            if (isPali(s, 0, r)) {
                return new StringBuilder(s.substring(r + 1)).reverse().toString() + s;
            }
        }
        return "";


    }
}

File: Python/214_ShortestPalindrome.py
def is_pali(s,l,r):
            while l <= r:
                if s[l] != s[r]:
                    return False
                l, r = l + 1, r-1
            return True
        
        for r in reversed(range(len(s))):
            if is_pali(s,0,r):
                return s[r+1:][::-1] + s
        
        return ""
